### PR TITLE
feat. Watch Later with API , Videos Page Filter

### PR DESCRIPTION
**Pull request by manojsarna**
### Features Added 
- [x] Watch Later Page with API & Functionality  Added
- [x] Loader Added
- [x] Toast Added with Toast Stacking in Toast Container
- [x] Watch Later Functionality with API Calls
- [x] Watch Later Icon Counts Added
- [x] Watch Later From Video Page Added
- [x] Videos Page With Filter Chips Added 
 
Deploy Link to Check : https://incomparable-gnome-0857d1.netlify.app/

### Files To Check

- [ ] [Card.jsx](https://github.com/manojsarna/manoj-smashtube-video-lib-react/compare/watchLaterPage-API-Functionality?expand=1#diff-fa5139f40dbeb869878fda25fffd4de5f0325f5c6fc57293a8f798cab43d57ed)
- [ ] [AddWatchLaterIcon.jsx](https://github.com/manojsarna/manoj-smashtube-video-lib-react/compare/watchLaterPage-API-Functionality?expand=1#diff-d1aac851521bf54522251b196fd7a11bce1b6aad9e0a9a580fdb30fee77cd632)
- [ ] [Toast.jsx](https://github.com/manojsarna/manoj-smashtube-video-lib-react/compare/watchLaterPage-API-Functionality?expand=1#diff-81829b4c685b73fd657a1c35eb8eaebd62da9c7ab4109ae52c3ef93972b0b85d)
- [ ] [ToastContainer.jsx](https://github.com/manojsarna/manoj-smashtube-video-lib-react/compare/watchLaterPage-API-Functionality?expand=1#diff-4dbef091da8591f0611f6f1d126b39c7c442fe625ced1d228bd6e82143b96a83)
- [ ] [toast-context.js](https://github.com/manojsarna/manoj-smashtube-video-lib-react/compare/watchLaterPage-API-Functionality?expand=1#diff-b84d19249ac16b37feeb8843502a2bad145979deffe1bf0e4cfc15c23aebd96a)
- [ ] [watch-later-context.js](https://github.com/manojsarna/manoj-smashtube-video-lib-react/compare/watchLaterPage-API-Functionality?expand=1#diff-e6119a96823dbdfc78dc09f8fa757d54fbe065cc2ff6e2de0a98ac14f16ce607)
- [ ] [WatchLater.jsx](https://github.com/manojsarna/manoj-smashtube-video-lib-react/compare/watchLaterPage-API-Functionality?expand=1#diff-f8fe38b2fbae7e8d6631596612db5213558c105380cc012b51ff0f890cdc5209)

### Ignore Rest of the Files

https://user-images.githubusercontent.com/45617406/160588452-13047b64-ea86-40e4-a571-b93023482b1d.mp4

### CODE REVIEW

**Comment by Cl0udCode** [comment]
good quality

File: src/context/playlists-context/playlists-context.js [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+        if (playlistsResponse.status === 200) {
+          setPlaylists(playlistsResponse.data.playlists);
+        }
+      } catch (error) {
+        console.error(error.response.data.errors);
+      }
+    })();
+  }, [encodedToken]);
+
+  const addToPlaylists = async (playlist) => {
```
Comment (b0llu) [inline src/context/playlists-context/playlists-context.js:34]: you can name this function as addPlaylist cause you are repeating same function name on line no 10.6

File: src/context/playlists-context/playlists-context.js [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+          payload: `Added to ${playlist.name}`,
+        });
+      }
+    } catch (error) {
+      console.error(error.response.data.errors);
+      dispatch({ type: "TOAST_ERROR", payload: error.response.data.errors });
+    }
+  };
+
+  const addToPlaylist = async (playlistId, video) => {
```
Comment (b0llu) [inline src/context/playlists-context/playlists-context.js:106]: same named function from above comment

**Review by b0llu — APPROVED** [review]
I suggest dont write same named functions again cause it may create some reviewing conflicts. Nothing else to suggest.

File: src/context/playlists-context/playlists-context.js (hunk 2 above)
Comment (manojsarna) [reply src/context/playlists-context/playlists-context.js:106]: AddToPlaylists : will add playlist

AddToPlaylist : will add video in that playlist

**Review by Veekshith-Thoram — APPROVED** [review]

File: src/context/watch-later-context/watch-later-context.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        {
+          headers: {
+            authorization: encodedToken,
+          },
+        }
+      );
+
+      if (response.status === 201) {
+        setWatchLater(response.data.watchlater);
+        console.log("in add watch later", response);
```
Comment (dev-ashishsethi) [inline src/context/watch-later-context/watch-later-context.js:47]: remove console log

File: src/reducers/toastReducer.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import { v4 as uuid } from "uuid";
+
+export const toastReducer = (state, action) => {
+  // console.log(state);
```
Comment (dev-ashishsethi) [inline src/reducers/toastReducer.js:4]: remove comments

File: src/reducers/toastReducer.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+      return [
+        ...state,
+        { _id: uuid(), type: "error", message: action.payload, show: true },
+      ];
+
+    case "TOAST_RESET":
+      return initialState;
+
+    default:
+      throw new Error(`Unhandled action type: ${action.type}`);
```
Comment (dev-ashishsethi) [inline src/reducers/toastReducer.js:29]: good logic and good use of custom toasts

File: src/pages/videos/Videos.jsx [hunk 6]
```diff
@@ -7,29 +7,61 @@ import { useState, useEffect } from "react";
...
+        <div className="sm-videos-categories-container">
+          {categories.map((category) => (
+            <button
+              key={category._id}
+              className={`sm-category-outline-btn   ${
+                category.category === curCat ? "sm-active" : ""
+              }`}
+              onClick={() => {
+                if (category.category === "all") {
+                  SetFVid(videos);
```
Comment (dev-ashishsethi) [inline src/pages/videos/Videos.jsx:46]: names should be self explanatory instead of SetFvid try something else

**Review by dev-ashishsethi — APPROVED** [review]
do check comments

**Pull request merged**